Tasks/XamariniOS/xamarinios.ts

```diff
@@ -1,75 +1,93 @@
 ﻿import path = require('path');
 import tl = require('vsts-task-lib/task');
 import sign = require('ios-signing-common/ios-signing-common');
+import msbuildhelpers = require('msbuildhelpers/msbuildhelpers');
 
-import {ToolRunner} from 'vsts-task-lib/toolrunner';
+import { ToolRunner } from 'vsts-task-lib/toolrunner';
 
 async function run() {
+
+    let codesignKeychain: string;
+    let profileToDelete: string;
+
     try {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
         // Get build inputs
-        var solutionPath = tl.getPathInput('solution', true, true);
-        var configuration = tl.getInput('configuration', true);
-        var clean = tl.getBoolInput('clean');
-        var args = tl.getInput('args');
-        var packageApp = tl.getBoolInput('packageApp');
-        var buildForSimulator = tl.getBoolInput('forSimulator');
-        var device = (buildForSimulator) ? 'iPhoneSimulator' : 'iPhone';
+        let solutionPath: string = tl.getPathInput('solution', true, true);
+        let configuration: string = tl.getInput('configuration', true);
+        let clean: boolean = tl.getBoolInput('clean');
+        let args: string = tl.getInput('args');
+        let packageApp: boolean = tl.getBoolInput('packageApp');
+        let buildForSimulator: boolean = tl.getBoolInput('forSimulator');
+        let device: string = (buildForSimulator) ? 'iPhoneSimulator' : 'iPhone';
         tl.debug('device: ' + device);
-        var xbuildLocation = tl.getInput('mdtoolLocation', false);
-        var cwd = tl.getInput('cwd');
-        let runNugetRestore : boolean = tl.getBoolInput('runNugetRestore');
-
-        // Get path to xbuild
-        var xbuildToolPath = undefined;
-        if (xbuildLocation) {
-            xbuildToolPath = path.join(xbuildLocation, 'xbuild');
-            tl.checkPath(xbuildToolPath, 'xbuild');
+        let cwd: string = tl.getInput('cwd');
+        let runNugetRestore: boolean = tl.getBoolInput('runNugetRestore');
+
+        // find the build tool path based on the build tool and location inputs
+        let buildTool: string = tl.getInput('buildTool');
+        let buildToolLocation: string = tl.getInput('mdtoolLocation', false);
+        let buildToolPath: string;
+        if (buildToolLocation) {
+            // location is specified
+            buildToolPath = buildToolLocation;
+            if (buildTool === 'xbuild' && !buildToolLocation.toLowerCase().endsWith('xbuild')) {
```

Comment:
It won't since we will have to run it under "Mono". For xbuild that feature never existed, so I left this to be on parity. For "msbuild.exe" it might be more useful to have that but I want to wait for user feedback on that feature.